frontend/src/component/common/Sticky/StickyProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState, useCallback, ReactNode, RefObject } from 'react';
+import { StickyContext } from './StickyContext';
+
+interface IStickyProviderProps {
+    children: ReactNode;
+}
+
+export const StickyProvider = ({ children }: IStickyProviderProps) => {
+    const [stickyItems, setStickyItems] = useState<RefObject<HTMLDivElement>[]>(
+        [],
+    );
+
+    const registerStickyItem = useCallback(
+        (item: RefObject<HTMLDivElement>) => {
+            setStickyItems((prevItems) => {
+                if (item && !prevItems.includes(item)) {
```

Comment:
that's a non trivial logic without tests

Reply:
Can you please clarify the part that you find non-trivial? The line on this comment is a simple "add if not added yet".
Anyways, added some tests: https://github.com/Unleash/unleash/pull/5088/commits/eb8a0aa8a9d042830bdd6eea88785c8345040ea4
Let me know if there are any specific tests you would like to see.

Reply:
Non trivial UI logic here is:
* checking includes
* adding to the end of the list
* sorting
* checking presence of both elements
* subtraction
If I change any of those I'd expect my tests to catch me by hand.
We can separate sorting algorithm from DOM and unit test the sorting independently.

Reply:
BTW if the code will tell it's difficult to do then I wouldn't proceed with the refactoring.